### DISCUSSION TITLE
RAG 방안논의: 유사도 기반 Retrieval이 아닌, 크롤링으로 직접 데이터 끌어오기

### DISCUSSION THREAD

**Discussion by hskhyl** [boostcampaitech7/level2-nlp-generationfornlp-nlp-07-lv3#4] [Ideas]
규태님과 논의 하였을 때에
총 3가지 방식으로 모델에 (넓은 의미에서의) RAG가 가능하다고 판단되었습니다.

1. DB를 구축해서(wiki.json 등 기타 자료 다 포함) 유사도 기반으로 Retrieval 하여 _LLM 기반 요약하여 input에 Hint로 제공_
2. Wikipedia API에서 제공하는 키워드 단위 검색기능 활용하여 얻은 데이터를 _LLM기반 요약하여 input에 Hint로 제공_
3. Wikipedia에 직접 request 날려서 크롤링한 데이터를 _LLM기반 요약하여 input에 Hint로 제공_

일단 2번까지는 무리 없이 될 것 같은데
3번은 운영진께 물어보고 좀 진행하도록 하겠습니다.

다른 Retrieval 관련 의견들 있으면 나눠주셔도 좋을 것 같습니다.

**Comment by hskhyl** [comment]
![image](https://github.com/user-attachments/assets/0e3ce4c1-dafc-4347-b0dc-c964a1a2fe91)

Test Dataset은 전혀 건드리지 않는 방향에서
Train dataset 기반 RAG DB를 구축하고 그 외에 공공기관 데이터를 좀 활용하면 좋을 것 같습니다 + 크롤링데이터

**Comment by DDUKDAE** [comment]
아니면 모델에게 힌트를 준다는 의미라면
retrieval을 해서 문서를 찾아서 주는 것도 좋을 것 같지만
gpt 4o모델 사용해서 테스트 데이터 문제 돌리고
해설 정도를 힌트로 넘겨주는 방식은 어때?
만약 이런거면 추론 과정에서 gpt 답변 생성하는 코드 넣어두면
룰에 크게 위반하지 않는 행동 아닌가?

**Reply by hskhyl** [reply]
좋은 의견인 것 같어 
마스터클래스때 한번 여쭤봐볼게!
